dimod/generators/constraints.py

```diff
@@ -91,9 +92,16 @@ def combinations(n, k, strength=1, vartype=BINARY):
     lbias = float(strength*(1 - 2*k))
     qbias = float(2*strength)
 
-    bqm = BinaryQuadraticModel.empty(BINARY)
-    bqm.add_variables_from(((v, lbias) for v in variables))
-    bqm.add_interactions_from(((u, v, qbias) for u, v in itertools.combinations(variables, 2)))
-    bqm.offset += strength*(k**2)
+    if not isinstance(n, int):
```

Comment:
this likely works for most cases that we care about, but we may as well be consistent with the old behaviour/exceptions. So
```
    if isinstance(n, abc.Collection):
        num_vars = len(n)
        variables = n
    else:
        num_vars = n
        try:
            variables = range(n)
        except TypeError:
            raise TypeError('n should be a collection or an integer')
```